### DISCUSSION TITLE
Trying to add new dialogue?

### DISCUSSION THREAD

**Discussion by fingerbangg** [zippity21/Stardew_Valley_Regression_Mod#63] [Q&A]
Sorry if this isn't the place to ask this, but I didn't know where else to ask. I wanted to add dialogue for Abigail just for my own fun, so I went into the files and added a new file under dialogue in the same format as every other file, and went to the content.json file to add her file the same way all the others are added, and it's just defaulting to her original dialogue. I did research and use the modding wiki dialogue section as a guide for writing the actual text as well, and when I launch the game there's no errors showing on the SMAPI so I don't think I'm doing anything particularly wrong in there, I just can't for the life of me figure out what I'm missing to make it actually show in game. Any help would be super appreciated!

Edit: Just in case it wasn't implied, I downloaded Visual Studios and did it that way (those instructions were also followed from the stardew mod wiki)

**Comment by ghost** [comment]
I'm happy to be a second pair of eyes if you're cool with attaching your Abigail.json.

I've been working on dialogue (slowly) and what's worked for me has been pretty much what you're doing, from the sound of things. Only difference I'm seeing is that instead of creating a fresh dialogue file, I use a copy of an existing NPC dialogue file, then modify it to suit my needs for a new NPC.

My full process has been:

1. Copying one of the existing NPC .json files in the Dialogue folder and pasting it back in.
2. Renaming it to the NPC I'm writing for (e.g. "Linus.json").
3. Opening the new .json in a text editor and updating the "LogName":"NPC name" and "Characters/Dialogue/NPC name" fields.
4. Deleting the rest of the contents and writing my own.
5. Add the Content Patcher call into the Content.json.

Here's a quick example I just whipped up with Abigail. This is the contents of Abigail.json:

```
{
	"Changes":[
		{
			"LogName":"Abigail",
			"Action":"EditData",
			"Target":"Characters/Dialogue/Abigail",
			"Entries":{				
  "spring_Sun4":"Smells like someone could use a change!$h",

  
			},
		},
	]
}
```

And this is all I'm adding to the Content.json:

```
	{
            "LogName": "Abigail",
            "Action": "Include",
            "FromFile": "Dialogue/NPCs/Abigail.json"
        },
```

And the result:

![image](https://user-images.githubusercontent.com/93164132/187034262-6a694582-35b6-42e8-98b5-e7db2f834cab.png)
